syntax = "proto3";
package plugin_version;




import "model/agent_admin/plugin.proto";
import "model/agent_admin/plugin_version.proto";

/*
GetPluginVersionV1请求
*/
message GetPluginVersionV1Request {
    /*
    插件Id
    */
    string id = 1 ;
    /*
    版本Id
    */
    string verId = 2 ;
}

/*
GetPluginVersionV1返回
*/
message GetPluginVersionV1Response {
    /*
    插件包版本名称
    */
    string versionName = 1 ;
    /*
    所属插件包
    */
    agent_admin.Plugin agentPlugin = 2 ;
    /*
    基线版本
    */
    agent_admin.PluginVersion baseVersion = 3 ;
    /*
    ID
    */
    string id = 4 ;
    /*
    仓库版本ID
    */
    string repoVersionId = 5 ;
    /*
    版本名称
    */
    string name = 6 ;
    /*
    备注
    */
    string memo = 7 ;
    /*
    所属插件ID
    */
    string pluginId = 8 ;
    /*
    创建者
    */
    string creator = 9 ;
    /*
    创建时间
    */
    int32 ctime = 10 ;
    /*
    修改时间
    */
    int32 mtime = 11 ;
}

/*
GetPluginVersionV1Api返回
*/
message GetPluginVersionV1ResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    plugin_version.GetPluginVersionV1Response data = 4 ;
}

